e-commerce/src/pages/catalog/catalog.tsx

```diff
@@ -1,12 +1,216 @@
+  const showMessageErrorSearch = () => {
+    setMessageError(true);
+    setTimeout(async () => {
+      setMessageError(false);
+      setCategoryId(categoryIdDefault);
+      const productsAllGet: IProductResponseCategory =
+        await getProductsSorted(categoryIdDefault);
+      const productsProps = formattedDataForCardInCategory(productsAllGet);
+      setProductCardProps(productsProps);
+    }, 3000);
+  };
+  const handleSearchPanel: OnClickType['onClick'] = async () => {
+    if (searchQuery.length > 1) {
+      const searchResponse = await searchProducts(searchQuery);
+      if (!searchResponse.total) {
+        showMessageErrorSearch();
+      }
+      const searchResponseProductProps =
+        formattedDataForCardInCategory(searchResponse);
+      setProductCardProps(searchResponseProductProps);
+      setSearchQuery('');
+    }
+  };
+  // Open product page
+  const openProduct = async (id: string) => {
+    if (id) {
+      navigate(`/product/${id}`);
+    }
+  };
+  // Handle Clicked product
+  const handlerProductChoose: (
+    event: React.MouseEvent<HTMLDivElement>
+  ) => void = async (event) => {
+    const productChoiceClick = event.currentTarget.getAttribute('data-id');
+    if (productChoiceClick) {
+      await openProduct(productChoiceClick);
+    }
+  };
+
+  // get products in filter
+  const getProductsFilter = useCallback((data: IProductResponseCategory) => {
+    if (!data.results[0]) {
+      showMessageErrorSearch();
+    }
+    const productsProps = formattedDataForCardInCategory(data);
+    setProductCardProps(productsProps);
+  }, []);
+
+  useEffect(() => {
+    const products = async () => {
+      try {
+        const productsAllGet: IProductResponseCategory =
+          await getProductsSorted(categoryIdDefault);
+        const productsProps = formattedDataForCardInCategory(productsAllGet);
+        setProductCardProps(productsProps);
+      } catch {
+        setError('Failed to fetch products. Please try again later.');
+      } finally {
+        setLoading(false);
+      }
+    };
+    const categories = async () => {
+      const categoriesAllGet: ICategoriesResponse = await getCategories();
+      const categoriesAllData = formattedDataForCategory(categoriesAllGet);
+      setCategoryProps(categoriesAllData);
+    };
+    products();
+    categories();
+  }, []);
+
+  if (loading) {
+    return <Spinner />;
+  }
+
+  if (error) {
+    return <div>{error}</div>;
+  }
+
   return (
-    <>
-      <h2 className="catalog free-page">Catalog</h2>
-      <WetFloor />
-    </>
+    <section className="catalog">
+      <h2 className="catalog-title">Catalog</h2>
+      <div className="catalog-wrapper">
+        <aside className="categories">
+          {categoryProps.map((category) => (
+            <Category
+              key={category.id}
+              name={category.name}
+              id={category.id}
+              onClick={handlerCategoryChoose}
+              isCurrent={category.id === categoryId}
```

Comment:
Кнопка с названием категории должна просто менять стейт выбранной категории (а рефетчинг продуктов происходит благодаря useEffect выше, с массивом зависимостей в котором есть стейт категории)
```suggestion
              onClick={() => setCategoryId(category.id)}
```